programs/lz4io.h

```diff
@@ -48,69 +48,77 @@ static const char nulmark[] = "nul";
 static const char nulmark[] = "/dev/null";
 #endif
 
+/* ************************************************** */
+/* ****************** Type Definitions ************** */
+/* ************************************************** */
+
+typedef struct LZ4IO_prefs_s LZ4IO_prefs_t;
+
+LZ4IO_prefs_t* LZ4IO_defaultPreferences(void);
+void LZ4IO_freePreferences(LZ4IO_prefs_t* prefs);
 
 /* ************************************************** */
 /* ****************** Functions ********************* */
 /* ************************************************** */
 
-int LZ4IO_compressFilename  (const char* input_filename, const char* output_filename, int compressionlevel);
-int LZ4IO_decompressFilename(const char* input_filename, const char* output_filename);
+int LZ4IO_compressFilename  (LZ4IO_prefs_t* prefs, const char* input_filename, const char* output_filename, int compressionlevel);
```

Comment:
Do you need the `LZ4IO_prefs_t` structure to be writable in these functions ?

Reply:
For all of them, it cannot be a `const LZ4IO_prefs_t*` due to the eventual calls to`LZ4IO_openDstFile` possibly modifying the `sparseFileSupport` field of the struct. Thus this does need to be writable.

That being said though, these can all be made `LZ4IO_prefs_t* const`s since the pointer should never changed (which will be done shortly).